### ISSUE TITLE
signature of model.predict_on_batch(inputs) different from readme

### ISSUE THREAD

**Issue by hendyadys** [fizyr/keras-retinanet#626]
i am trying to train keras-retinanet, but the outputs of model.predict_on_batch(inputs) seem different from the readme and https://github.com/fizyr/keras-retinanet/blob/master/examples/ResNet50RetinaNet.ipynb

it is supposed to be 
`boxes, scores, labels = model.predict_on_batch(inputs)`
but seems to be 
`boxes, nms_classification  = model.predict_on_batch(inputs)`

this would be fine if nms_classification was correct, but using the old definitions that i dug up 
```
predicted_labels = np.argmax(nms_classification[0, :, :], axis=1)
scores = nms_classification[0, np.arange(nms_classification.shape[1]), predicted_labels]
sig_indices = np.where(scores > 0.5)[0]
print(img_name, sig_indices, boxes.shape)
```
gives nonsense
('000000008021.jpg', array([159242, 159245, 159251, 159254, 159394, 159395, 159397, 159398,
       159403, 159404, 159407, 159542, 159547, 159548, 159551, 159557,
       159560, 159704, 160671, 160674]), (1, 160956, 4),
      )
and box coordinates even have negative values.

what am i missing?
i am on Keras 2.2.0 and tensorflow 1.7.0 and up to date with keras-retinanet-master

i tried training on coco and pascal, as well as my own dataset with csv
python /keras-retinanet-master/keras_retinanet/bin/train.py --gpu 0 --backbone resnet50 --epochs 10 --batch-size 1 --snapshot-path ./snapshots/pascal pascal /path/to/pascal

python /keras-retinanet-master/keras_retinanet/bin/train.py --gpu 1 --backbone resnet50 --epochs 10 --batch-size 1 --no-weights --snapshot-path /snapshots/my_data csv /path/to/mydata.csv /path/to/mydata_classes.csv

**Comment by hgaiser** [comment]
I would guess that you're using a training model to do inference. The difference between the two is that an inference model adds some utility functions to, for instance, convert the regression values to bounding boxes and to apply NMS. Check the README for more information.

**Comment by hendyadys** [comment]
thanks that was the problem
